ibridges/interactive.py

```diff
@@ -39,35 +38,60 @@ def interactive_auth(password: Optional[str] = None,
         print(f'File not found: {irods_env_path}')
         raise FileNotFoundError
 
-    if os.path.exists(Path(os.path.expanduser("~")).joinpath(".irods", ".irodsA")):
-        try:
-            session = Session(irods_env_path)
-            return session
-        except IndexError:
-            print('INFO: The cached password in ~/.irods/.irodsA has been corrupted')
-        except ValueError:
-            print('INFO: The cached password in ~/.irods/.irodsA is wrong.')
+    if os.path.exists(Path(os.path.expanduser("~")).joinpath(".irods", ".irodsA")) and \
+            password is None:
+        session = _from_pw_file(irods_env_path)
 
-    with open(irods_env_path, "r", encoding="utf-8") as f:
-        ienv = json.load(f)
     if password is not None:
+        session = _from_password(irods_env_path, password)
+
+    if session is not None:
+        return session
+
+    # If provided passwords in file or on prompt were wrong
+    n_tries = 0
+    success = False
+    while not success and n_tries < 3:
+        password = getpass("Your iRODS password: ")
         try:
-            session = Session(ienv, password=password)
+            session = Session(irods_env=irods_env_path, password=password)
             session.write_pam_password()
+            success = True
             return session
-        except ValueError:
+        except ValueError as e:
+            print(repr(e))
             print('INFO: The provided password is wrong.')
-    else:
-        n_tries = 0
-        success = False
-        while not success and n_tries < 3:
-            password = getpass("Your iRODS password: ")
-            try:
-                session = Session(irods_env=ienv, password=password)
-                session.write_pam_password()
-                success = True
-                return session
-            except ValueError:
-                print('INFO: The provided password is wrong.')
-                n_tries+=1
+            n_tries+=1
     raise ValueError("Connection to iRODS could not be established.")
+
+def _from_pw_file(irods_env_path):
+    try:
+        session = Session(irods_env_path)
+        return session
+    except LoginError as e:
+        print(f'INFO: {e.args}')
+        # environment.json is not setup correctly -> stop
+        if 'not set correctly in irods_environment.json' in e.args[0]:
+            raise
+        # non-iRODS key found
+        if 'Unexpected value in irods_environment; ' in e.args:
+            raise(ValueError('Non-iRODS key or value found in environment.json')) from e
```

Comment:
```suggestion
            raise ValueError('Non-iRODS key or value found in environment.json') from e
```
raise doesn't need the `()`.